sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs

```diff
@@ -11,16 +16,59 @@ namespace Azure.Core.TestFramework
     ///   Represents the ambient environment in which the test suite is
     ///   being run.
     /// </summary>
-    public partial class TestEnvironment
+    public abstract class TestEnvironment
     {
-        private TokenCredential _credential;
+        private static readonly string RepositoryRoot;
         private readonly string _prefix;
 
-        public TestEnvironment(string serviceName)
+        private TokenCredential _credential;
+        private TestRecording _recording;
+
+        private readonly Dictionary<string, string> _environmentFile = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+
+        protected TestEnvironment(string serviceName)
         {
             _prefix = serviceName.ToUpperInvariant() + "_";
+            if (RepositoryRoot == null)
+            {
+                throw new InvalidOperationException("Unexpected error, repository root not found");
+            }
+
+            var testEnvironmentFile = Path.Combine(RepositoryRoot, "sdk", serviceName, "test-resources.json.env");
+            if (File.Exists(testEnvironmentFile))
+            {
+                var json = JsonDocument.Parse(
+                    ProtectedData.Unprotect(File.ReadAllBytes(testEnvironmentFile), null, DataProtectionScope.CurrentUser)
+                );
+
+                foreach (var property in json.RootElement.EnumerateObject())
+                {
+                    _environmentFile[property.Name] = property.Value.GetString();
+                }
+            }
         }
 
+        static TestEnvironment()
+        {
+            // Traverse parent directories until we find an "artifacts" directory
+            // parent of that would become a repo root for test environment resolution purposes
+            var directoryInfo = new DirectoryInfo(Assembly.GetExecutingAssembly().Location);
+
+            while (directoryInfo.Name != "artifacts")
```

Comment:
This is depending on our current structure and naming, it might be worth considering another configuration mechanism... dare I suggest an environment variable :)

Reply:
If we change naming I'll fix this code. Don't think it happens often enough to invent something fancy.